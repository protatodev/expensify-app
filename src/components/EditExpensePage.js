import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense } from '../actions/expenses';
import { startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit = expense => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');  
  };

  onRemoveExpense = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button
        onClick={this.onRemoveExpense}
        >Remove</button>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  startRemoveExpense: id => dispatch(startRemoveExpense(id)) 
});

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(p => p.id === props.match.params.id)
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);