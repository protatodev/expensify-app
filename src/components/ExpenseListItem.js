import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({description, amount, createdAt, id }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>Cost: ${amount}</p>
    <p>Date Created: {createdAt}</p>
    <hr/>
  </div>
);

export default ExpenseListItem;