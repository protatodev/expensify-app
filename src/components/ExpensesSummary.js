import React from 'react';
import { connect } from 'react-redux';
import expensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
  <div>
    <p>Viewing {props.expenseCount} expense(s) totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}</p>
  </div>
);

const mapStateToProps = (state) => ({
  expenseCount: selectExpenses(state.expenses, state.filters).length,
  expensesTotal: expensesTotal(selectExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpensesSummary);

