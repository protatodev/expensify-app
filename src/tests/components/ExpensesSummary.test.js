import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';
import React from 'react';

test('should set up snapshot for ExpensesSummary component', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={0}/>);
  expect(wrapper).toMatchSnapshot();
})