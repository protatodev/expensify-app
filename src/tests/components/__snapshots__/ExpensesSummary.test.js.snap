// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should set up snapshot for ExpensesSummary component 1`] = `
<div>
  <p>
    Viewing 
    2
     expense(s) totalling 
    $0.00
  </p>
</div>
`;
